tests/Functional/ControllerTest.php

```diff
@@ -0,0 +1,211 @@
+        $request->headers->set('Content-Type', MimeTypes::APPLICATION_JSON);
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertJsonStringEqualsJsonString($submitted, $response->getContent());
+    }
+
+    public function testReconstitutesMultipleFromRequest(): void
+    {
+        $submitted = <<<JSON
+            [
+                {"from":"Jenny"},
+                {"from":"Forest"}
+            ]
+        JSON;
+
+        $request = Request::create('reconstitutesMany', Request::METHOD_POST, [], [], [], [], $submitted);
+        $request->headers->set('Content-Type', MimeTypes::APPLICATION_JSON);
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertJsonStringEqualsJsonString($submitted, $response->getContent());
+    }
+
+    private function deleteTempDir()
+    {
+        if (!file_exists($dir = sys_get_temp_dir() . '/' . Kernel::VERSION . '/CrudControllerTestKernel')) {
+            return;
+        }
+
+        $fs = new Filesystem();
+        $fs->remove($dir);
+    }
+}
+
+/** @internal  */
+final class TestKernel extends Kernel
+{
+    use MicroKernelTrait;
+
+
+    public function __construct()
+    {
+        parent::__construct('test', false);
+    }
+
+    public function registerBundles(): array
+    {
+        return [
+            new FrameworkBundle(),
+            new ViolinesRestBundle(),
+        ];
+    }
+
+    protected function configureRoutes(RouteCollectionBuilder $routes): void
+    {
+        $routes->addRoute(new Route('/returnsOne', [
+            '_controller' => HugController::class . '::returnsOne',
+        ]));
+
+        $routes->addRoute(new Route('/returnsMany', [
+            '_controller' => HugController::class . '::returnsMany',
+        ]));
+
+        $routes->addRoute(new Route('/reconstitutesOne', [
+            '_controller' => HugController::class . '::reconstitutesOne',
+        ]));
+
+        $routes->addRoute(new Route('reconstitutesMany', [
+            '_controller' => HugController::class . '::reconstitutesMany',
+        ]));
+    }
+
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    {
+    }
+
+    public function getCacheDir(): string
+    {
+        return sys_get_temp_dir() . '/' . Kernel::VERSION . '/CrudControllerTestKernel/cache/' . $this->environment;
+    }
+
+    public function getLogDir(): string
+    {
+        return sys_get_temp_dir() . '/' . Kernel::VERSION . '/CrudControllerTestKernel/logs';
+    }
+}
+
+final class HugController
+{
+    public function returnsOne(): Hug
+    {
+        $fromForest = new Hug();
+        $fromForest->from = 'Forest';
+
+        return $fromForest;
+    }
+
+    public function returnsMany(): array
+    {
+        $fromJenny = new Hug();
```

Comment:
I would suggest to use the constructor:
`$fromJenny = new Hug('Jenny');`

btw. I like the ref to Forest Gump 👍

Reply:
done :-)